build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -15,30 +15,135 @@ const scriptLocation = '.dart_tool/build/build.dart';
-      .where((config) => config.builderDefinitions.isNotEmpty)
-      .map((config) => refer('print').call([
-            literalString('I would run: \'${config.packageName} - '
-                '${config.builderDefinitions.keys.toList()}')
-          ]).statement);
-  final library = new File((b) => b.body.addAll([
-        new Method.returnsVoid((b) => b
-          ..name = 'main'
-          ..body = new Block((b) => b..statements.addAll(printStatements)))
-      ]));
-  final emitter = new DartEmitter();
+      (await Future.wait(packageGraph.orderedPackages.map(_packageBuildConfig)))
+          .where((c) => c.builderDefinitions.isNotEmpty);
+  final library = new File(
+      (b) => b.body.addAll([_findBuildActions(buildConfigs), _main()]));
+  final emitter = new DartEmitter(new Allocator.simplePrefixing());
   return new DartFormatter().format('${library.accept(emitter)}');
 }
 
+Method _findBuildActions(Iterable<BuildConfig> buildConfigs) =>
+    new Method((b) => b
+      ..name = '_buildActions'
+      ..requiredParameters.add(new Parameter((b) => b
+        ..name = 'graph'
+        ..type = _packageGraph))
+      ..returns = _buildActions
+      ..body = new Block((b) => b
+        ..statements.addAll([
+          _buildActions.newInstance([]).assignVar('actions').statement,
+          // TODO - Pass actual arguments
+          literalList([], new TypeReference((b) => b..symbol = 'String'))
+              .assignVar('args')
+              .statement,
+        ]
+          ..addAll(_addBuildActions(buildConfigs))
+          ..add(refer('actions').returned.statement))));
+
+Method _main() => new Method((b) => b
+  ..name = 'main'
+  ..modifier = MethodModifier.async
+  ..body = new Block((b) => b
+    ..statements.addAll([
+      refer('_buildActions')
+          .call([_packageGraph.newInstanceNamed('forThisPackage', [])])
+          .assignVar('actions')
+          .statement,
+      refer('watch', 'package:build_runner/build_runner.dart')
+          .call([refer('actions')], {'writeToCache': literalTrue})
+          .awaited
+          .assignVar('handler')
+          .statement,
+      refer('serve', 'package:shelf/shelf_io.dart')
+          .call([
+            refer('handler')
+                .property('handlerFor')
+                .call([literalString('web')]),
+            literalString('localhost'),
+            literal(8000)
+          ])
+          .awaited
+          .assignVar('server')
+          .statement,
+      refer('handler')
+          .property('buildResults')
+          .property('drain')
+          .call([])
+          .awaited
+          .statement,
+      refer('server').property('close').call([]).awaited.statement
+    ])));
+
+Iterable<Code> _addBuildActions(Iterable<BuildConfig> configs) {
+  var statements = <Code>[];
+  for (var config in configs) {
+    var varName = 'buildersFor${config.packageName}';
+    statements.addAll([
+      _packageBuilders(config, varName),
+      refer('graph')
```

Comment:
consider naming this `packageGraph` to disambiguate

Reply:
Done